tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,204 @@
+package charts
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	bv1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type BackupTestSuite struct {
+	suite.Suite
+	client              *rancher.Client
+	session             *session.Session
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+	chartFeatureOptions *charts.RancherBackupOpts
+	S3Client            *s3.Client
+}
+
+func (b *BackupTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BackupTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	b.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(b.T(), err)
+
+	b.client = client
+
+	broConfig := new(charts.Config)
+	config.LoadConfig(charts.ConfigurationFileKey, broConfig)
+
+	project, err := projects.GetProjectByName(client, cluster, clusterProject)
+	require.NoError(b.T(), err)
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(b.T(), clusterName, "Cluster name to install is not set")
+
+	// Get cluster meta
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(b.T(), err)
+
+	// Get latest version of Rancher Backup chart
+	latestBackupVersion, err := client.Catalog.GetLatestChartVersion(backupChartName, catalog.RancherChartRepo)
+	require.NoError(b.T(), err)
+
+	b.chartInstallOptions = &charts.InstallOptions{
+		Cluster:   cluster,
+		Version:   latestBackupVersion,
+		ProjectID: project.ID,
+	}
+	b.chartFeatureOptions = &charts.RancherBackupOpts{
```

Comment:
Please also move this out of setup as we are not going to declare this in the global struct.